extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonSerializationUtils.java

```diff
@@ -0,0 +1,65 @@
+package io.quarkus.resteasy.reactive.jackson.deployment.processor;
+
+import org.jboss.jandex.Type;
+import org.jboss.jandex.Type.Kind;
+
+import io.quarkus.gizmo.BytecodeCreator;
+import io.quarkus.gizmo.ResultHandle;
+
+final class JacksonSerializationUtils {
+
+    private static boolean DEFAULT_BOOLEAN;
+    private static char DEFAULT_BYTE;
+    private static char DEFAULT_CHAR;
+    private static double DEFAULT_DOUBLE;
+    private static float DEFAULT_FLOAT;
+    private static int DEFAULT_INT;
+    private static long DEFAULT_LONG;
+    private static short DEFAULT_SHORT;
+
+    private JacksonSerializationUtils() {
+    }
+
+    static boolean isBoxedPrimitive(Type type) {
+        return isBoxedPrimitive(type.name().toString());
+    }
+
+    static boolean isBoxedPrimitive(String typeName) {
+        return "java.lang.Character".equals(typeName) || "java.lang.Short".equals(typeName)
+                || "java.lang.Integer".equals(typeName) || "java.lang.Long".equals(typeName)
+                || "java.lang.Float".equals(typeName) || "java.lang.Double".equals(typeName)
+                || "java.lang.Boolean".equals(typeName);
+    }
```

Comment:
@Ladicek I was wondering if we should have a util in Jandex for this. I couldn't find one but maybe there is one?

Reply:
👍🏽 

That could probably be used in a lot of places.